app/app/static/custom/ourscript.js

```diff
@@ -387,3 +412,202 @@ const requestCSV = async (data) => {
+		var data = {series: unitsCountArray};
+		
+		var sum = function(a, b) {return a + b};
+
+		// Set options for unit chart
+		var options = {
+			labelInterpolationFnc: function(value, idx) {
+				var percentage = Math.round(value / data.series.reduce(sum) * 100) + '%';
+				return unitsArray[idx] + ' ' + percentage;
+			}
+		};
+		
+		// Set responsive options for unit chart
+		var responsiveOptions = [
+			['screen and (min-width: 640px)', {
+				chartPadding: 30,
+				labelOffset: 100,
+				labelDirection: 'explode',
+			}],
+			['screen and (min-width: 1024px)', {
+				labelOffset: 50,
+				chartPadding: 20
+			}]
+		];
+		
+		// Create unit graph
+		new Chartist.Pie('#unitsPieChart', data, options, responsiveOptions);
+	}
+	// Display blank pie graph saying "No data for this week"
+	else {
+		new Chartist.Pie('#unitsPieChart', {series: [100]}, {
+			labelInterpolationFnc: function(value) {
+			  return "No data for this week"
+			}
+		  });
+	}
+
+	//----------------------------STAFF PIE GRAPH----------------------------
+	// Get and set data for staff chart
+	let staffPieValues = inputData["staffPieValues"];
+	const staffTypes = ['STUDYSmarter', 'Librarians']
+
+	// Only displays data if there was any that week
+	if(staffPieValues['STUDYSmarter'] > 0 || staffPieValues['Librarian'] > 0) {
+		var data = {
+			series: [staffPieValues['STUDYSmarter'], staffPieValues['Librarian']]
+		};
+		
+		// Set options for staff chart
+		var options = {
+			labelInterpolationFnc: function(value, idx) {
+				var percentage = Math.round(value / data.series.reduce(sum) * 100) + '%';
+				return staffTypes[idx] + ' ' + percentage;
+			}
+		};
+		
+		// Set responsive options for staff chart
+		var responsiveOptions = [
+			['screen and (min-width: 640px)', {
+				chartPadding: 30,
+				labelOffset: 100,
+				labelDirection: 'explode',
+			}],
+			['screen and (min-width: 1024px)', {
+				labelOffset: 50,
+				chartPadding: 20
+			}]
+		];
+		
+		// Create staff graph
+		new Chartist.Pie('#unitsStaffChart', data, options, responsiveOptions);
+	}
+	// Display blank pie graph saying "No data for this week"
+	else {
+		new Chartist.Pie('#unitsStaffChart', {series: [100]}, {
+			labelInterpolationFnc: function(value) {
+			  return "No data for this week"
+			}
+		  });
+	}
+}
+
+$('#dateSubmitForAnalytics').on('click', function(e) {
+	// Get values from page
+	const dateSubmitted = document.getElementById("date").value;
+
+	var date = new Date(dateSubmitted);
+
+	// Gets the most recently passed Sunday and the next upcomming Saturday
+	var firstDay = new Date(date.setDate(date.getDate() - date.getDay())).toJSON().slice(0,10);
+	var lastDay = new Date(date.setDate(date.getDate() - date.getDay() + 6)).toJSON().slice(0,10);
+
+	// Format the dates to the appropriate dateTime format
+	const startTime = `${firstDay} 00:00:00.0000000`;
+	const endTime = `${lastDay} 23:59:59.999999`;
+	// Send html request
+	createDataChart({
+		startTime,
+		endTime
+	})
+})
```

Comment:
As suggestion above, this repetition can be fixed to
``` 
$('#dateSubmitForAnalytics').on('click', function(e) {
	// Get values from page
	const dateSubmitted = document.getElementById("date").value;

	var date = new Date(dateSubmitted);
        genChartsForGivenWeek(date);
}

Reply:
Good pick up, will do